test/integration/scheduler/util.go

```diff
@@ -185,58 +170,61 @@ func initTestSchedulerWithOptions(
 	} else {
 		podInformer = context.informerFactory.Core().V1().Pods()
 	}
-
-	context.schedulerConfigArgs = createConfiguratorArgsWithPodInformer(
-		v1.DefaultSchedulerName, context.clientSet, podInformer, context.informerFactory, pluginRegistry, plugins,
-		pluginConfig, context.stopCh)
-	configFactory := factory.NewConfigFactory(context.schedulerConfigArgs)
-
 	var err error
-
-	if policy != nil {
-		context.schedulerConfig, err = configFactory.CreateFromConfig(*policy)
-	} else {
-		context.schedulerConfig, err = configFactory.Create()
-	}
-
-	if err != nil {
-		t.Fatalf("Couldn't create scheduler config: %v", err)
-	}
-
-	// set DisablePreemption option
-	context.schedulerConfig.DisablePreemption = disablePreemption
 	eventBroadcaster := events.NewBroadcaster(&events.EventSinkImpl{
 		Interface: context.clientSet.EventsV1beta1().Events(""),
 	})
-	context.schedulerConfig.Recorder = eventBroadcaster.NewRecorder(
+	recorder := eventBroadcaster.NewRecorder(
 		legacyscheme.Scheme,
 		v1.DefaultSchedulerName,
 	)
-
-	context.scheduler = scheduler.NewFromConfig(context.schedulerConfig)
-
-	scheduler.AddAllEventHandlers(context.scheduler,
-		v1.DefaultSchedulerName,
+	var algorithmSrc schedulerconfig.SchedulerAlgorithmSource
+	if policy != nil {
+		algorithmSrc = createAlgorithmSourceFromPolicy(policy, context.clientSet)
+	} else {
+		provider := schedulerconfig.SchedulerDefaultProviderName
+		algorithmSrc = schedulerconfig.SchedulerAlgorithmSource{
+			Provider: &provider,
+		}
+	}
+	opts = append(opts, scheduler.WithBindTimeoutSeconds(600))
```

Comment:
@alculquicondor I think this should be changed to `opts = append([]scheduler.Option{scheduler.WithBindTimeoutSeconds(600)}, opts...)`. In order to enable other callers to set BindTimeoutSeconds and still work as expected. Is that correct?

Is there a specific reason for using 600 here and not the default value?

Reply:
That's a good idea. I assume the reason could be to reduce potential flakiness.